src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,136 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INDEX_NOT_FOUND = -1;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = new Object[DEFAULT_CAPACITY];
+    }
+
+    private void grow() {
+        elementData = Arrays.copyOf(elementData, elementData.length + (elementData.length >> 1));
```

Comment:
Common mistakes:
Use System.arraycoppy() to move your array elements.

Reply:
We will create new array in grow method. Think we still can use Arrays.copyOf() 'cause it return same new array with newLength. There are no taboo for using Arrays.copyOf() in Common mistakes.